exchanges/bitstamp/bitstamp.go

```diff
@@ -86,6 +83,40 @@ func (b *Bitstamp) GetFee(ctx context.Context, feeBuilder *exchange.FeeBuilder)
 	return fee, nil
 }
 
+// GetTradingFee returns a trading fee based on a currency
+func (b *Bitstamp) getTradingFee(ctx context.Context, feeBuilder *exchange.FeeBuilder) (float64, error) {
+	fees, err := b.GetAccountTradingFee(ctx, feeBuilder.Pair)
+	if err != nil {
+		return 0, err
+	}
+	if len(fees) != 1 {
+		return 0, errors.New("did not receive exactly one fee from api")
+	}
+	fee := fees[0].Taker
+	if feeBuilder.IsMaker {
+		fee = fees[0].Maker
+	}
+	return fee / 100 * feeBuilder.PurchasePrice * feeBuilder.Amount, nil
+}
+
+// GetAccountTradingFee returns a TradingFee for a pair
+func (b *Bitstamp) GetAccountTradingFee(ctx context.Context, pair currency.Pair) ([]MakerTakerFees, error) {
+	path := bitstampAPITradingFees + "/" + strings.ToLower(pair.String())
+
+	var resp TradingFees
+	err := b.SendAuthenticatedHTTPRequest(ctx, exchange.RestSpot, path, true, nil, &resp)
+	return resp.Fees, err
+}
+
+// GetAccountTradingFees returns a slice of TradingFee
+func (b *Bitstamp) GetAccountTradingFees(ctx context.Context) ([]TradingFees, error) {
```

Comment:
This endpoint returns errors, please add a test
```
Bitstamp raw response: [{"currency_pair": "1incheur", "market": "1incheur", "fees": {"maker": "0.000", "taker": "0.000"}}, {"currency_pair": "1inchusd", "market": "1inchusd", "fees": {"maker": "0.000", "taker": "0.000"}}, {"currency_pair": "aavebtc", "market": "aavebtc", "fees": {"maker": "0.000", "taker": "0.000"}}, {"currency_pair": "aaveeur", "market": "aaveeur", "fees": {"maker": "0.000", "taker": "0.000"}}, {"currency_pair": "aaveusd", "market": "aaveusd", "fees": {"maker": "0.000", "taker": "0.000"}}, {"currency_pair": "adabtc", "market": "adabtc", "fees": {"maker": "0.000", "taker": "0.000"}}, {"currency_pair": "adaeur", "market": "adaeur", "fees": {"maker": "0.000", "taker": "0.000"}}, {"currency_pair": "adausd", "market": "adausd", "fees": {"maker": "0.000", "taker": "0.000"}}, {"currency_pair": "algobtc", "market": "algobtc", "fees": {"maker": "0.000", "taker": "0.000"}}, {"currency_pair": "algoeur", "market": "algoeur", "fees": {"maker": "0.000", "taker": "0.000"}}, {"currency_pair": "algousd", "market": "algousd", "fees": {"maker": "0.000", "taker": "0.000"}}, {"currency_pair": "alphaeur", "market": "alphaeur", "fees": {"maker": "0.000", "taker": "0.000"}}, {"currency_pair": "alphausd", "market": "alphausd", "fees": {"maker": "0.000", "taker": "0.000"}}, {"currency_pair": "ampeur", "market": "ampeur", "fees": {"maker": "0.000", "taker": "0.000"}}, {"currency_pair": "ampusd", "market": "ampusd", "fees": {"maker": "0.000", "taker": "0.000"}}, {"currency_pair": "anteur", "market": "anteur", "fees": {"maker": "0.000", "taker": "0.000"}}, {"currency_pair": "antusd", "market": "antusd", "fees": {"maker": "0.000", "taker": "0.000"}}, {"currency_pair": "apeeur", "market": "apeeur", "fees": {"maker": "0.000", "taker": "0.000"}}, {"currency_pair": "apeusd", "market": "apeusd", "fees": {"maker": "0.000", "taker": "0.000"}}, {"currency_pair": "audiobtc", "market": "audiobtc", "fees": {"maker": "0.000", "taker": "0.000"}}, {"currency_pair": "audioeur", "market": "audioeur", "fees": {"maker": "0.000", "taker": "0.000"}}, {"currency_pair": "audiousd", "market": "audiousd", "fees": {"maker": "0.000", "taker": "0.000"}}, {"currency_pair": "avaxeur", "market": "avaxeur", "fees": {"maker": "0.000", "taker": "0.000"}}, {"currency_pair": "avaxusd", "market": "avaxusd", "fees": {"maker": "0.000", "taker": "0.000"}}, {"currency_pair": "axseur", "market": "axseur", "fees": {"maker": "0.000", "taker": "0.000"}}, {"currency_pair": "axsusd", "market": "axsusd", "fees": {"maker": "0.000", "taker": "0.000"}}, {"currency_pair": "bandeur", "market": "bandeur", "fees": {"maker": "0.000", "taker": "0.000"}}, {"currency_pair": "bandusd", "market": "bandusd", "fees": {"maker": "0.000", "taker": "0.000"}}, {"currency_pair": "bateur", "market": "bateur", "fees": {"maker": "0.000", "taker": "0.000"}}, {"currency_pair": "batusd", "market": "batusd", "fees": {"maker": "0.000", "taker": "0.000"}}, {"currency_pair": "bchbtc", "market": "bchbtc", "fees": {"maker": "0.000", "taker": "0.000"}}, {"currency_pair": "bcheur", "market": "bcheur", "fees": {"maker": "0.000", "taker": "0.000"}}, {"currency_pair": "bchusd", "market": "bchusd", "fees": {"maker": "0.000", "taker": "0.000"}}, {"currency_pair": "btceur", "market": "btceur", "fees": {"maker": "0.000", "taker": "0.000"}}, {"currency_pair": "btcgbp", "market": "btcgbp", "fees": {"maker": "0.000", "taker": "0.000"}}, {"currency_pair": "btcpax", "market": "btcpax", "fees": {"maker": "0.000", "taker": "0.000"}}, {"currency_pair": "btcusd", "market": "btcusd", "fees": {"maker": "0.000", "taker": "0.000"}}, {"currency_pair": "btcusdc", "market": "btcusdc", "fees": {"maker": "0.000", "taker": "0.000"}}, {"currency_pair": "btcusdt", "market": "btcusdt", "fees": {"maker": "0.000", "taker": "0.000"}}, {"currency_pair": "chzeur", "market": "chzeur", "fees": {"maker": "0.000", "taker": "0.000"}}, {"currency_pair": "chzusd", "market": "chzusd", "fees": {"maker": "0.000", "taker": "0.000"}}, {"currency_pair": "compeur", "market": "compeur", "fees": {"maker": "0.000", "taker": "0.000"}}, {"currency_pair": "compusd", "market": "compusd", "fees": {"maker": "0.000", "taker": "0.000"}}, {"currency_pair": "crveur", "market": "crveur", "fees": {"maker": "0.000", "taker": "0.000"}}, {"currency_pair": "crvusd", "market": "crvusd", "fees": {"maker": "0.000", "taker": "0.000"}}, {"currency_pair": "ctsieur", "market": "ctsieur", "fees": {"maker": "0.000", "taker": "0.000"}}, {"currency_pair": "ctsiusd", "market": "ctsiusd", "fees": {"maker": "0.000", "taker": "0.000"}}, {"currency_pair": "cvxeur", "market": "cvxeur", "fees": {"maker": "0.000", "taker": "0.000"}}, {"currency_pair": "cvxusd", "market": "cvxusd", "fees": {"maker": "0.000", "taker": "0.000"}}, {"currency_pair": "daiusd", "market": "daiusd", "fees": {"maker": "0.000", "taker": "0.000"}}, {"currency_pair": "dgldeur", "market": "dgldeur", "fees": {"maker": "0.000", "taker": "0.000"}}, {"currency_pair": "dgldusd", "market": "dgldusd", "fees": {"maker": "0.000", "taker": "0.000"}}, {"currency_pair": "dogeeur", "market": "dogeeur", "fees": {"maker": "0.000", "taker": "0.000"}}, {"currency_pair": "dogeusd", "market": "dogeusd", "fees": {"maker": "0.000", "taker": "0.000"}}, {"currency_pair": "doteur", "market": "doteur", "fees": {"maker": "0.000", "taker": "0.000"}}, {"currency_pair": "dotusd", "market": "dotusd", "fees": {"maker": "0.000", "taker": "0.000"}}, {"currency_pair": "dydxeur", "market": "dydxeur", "fees": {"maker": "0.000", "taker": "0.000"}}, {"currency_pair": "dydxusd", "market": "dydxusd", "fees": {"maker": "0.000", "taker": "0.000"}}, {"currency_pair": "enjeur", "market": "enjeur", "fees": {"maker": "0.000", "taker": "0.000"}}, {"currency_pair": "enjusd", "market": "enjusd", "fees": {"maker": "0.000", "taker": "0.000"}}, {"currency_pair": "enseur", "market": "enseur", "fees": {"maker": "0.000", "taker": "0.000"}}, {"currency_pair": "ensusd", "market": "ensusd", "fees": {"maker": "0.000", "taker": "0.000"}}, {"currency_pair": "eth2eth", "market": "eth2eth", "fees": {"maker": "0.000", "taker": "0.000"}}, {"currency_pair": "ethbtc", "market": "ethbtc", "fees": {"maker": "0.000", "taker": "0.000"}}, {"currency_pair": "etheur", "market": "etheur", "fees": {"maker": "0.000", "taker": "0.000"}}, {"currency_pair": "ethgbp", "market": "ethgbp", "fees": {"maker": "0.000", "taker": "0.000"}}, {"currency_pair": "ethpax", "market": "ethpax", "fees": {"maker": "0.000", "taker": "0.000"}}, {"currency_pair": "ethusd", "market": "ethusd", "fees": {"maker": "0.000", "taker": "0.000"}}, {"currency_pair": "ethusdc", "market": "ethusdc", "fees": {"maker": "0.000", "taker": "0.000"}}, {"currency_pair": "ethusdt", "market": "ethusdt", "fees": {"maker": "0.000", "taker": "0.000"}}, {"currency_pair": "euroceur", "market": "euroceur", "fees": {"maker": "0.000", "taker": "0.000"}}, {"currency_pair": "eurocusdc", "market": "eurocusdc", "fees": {"maker": "0.000", "taker": "0.000"}}, {"currency_pair": "eurteur", "market": "eurteur", "fees": {"maker": "0.000", "taker": "0.000"}}, {"currency_pair": "eurtusd", "market": "eurtusd", "fees": {"maker": "0.000", "taker": "0.000"}}, {"currency_pair": "eurusd", "market": "eurusd", "fees": {"maker": "0.000", "taker": "0.000"}}, {"currency_pair": "feteur", "market": "feteur", "fees": {"maker": "0.000", "taker": "0.000"}}, {"currency_pair": "fetusd", "market": "fetusd", "fees": {"maker": "0.000", "taker": "0.000"}}, {"currency_pair": "flreur", "market": "flreur", "fees": {"maker": "0.000", "taker": "0.000"}}, {"currency_pair": "flrusd", "market": "flrusd", "fees": {"maker": "0.000", "taker": "0.000"}}, {"currency_pair": "ftmeur", "market": "ftmeur", "fees": {"maker": "0.000", "taker": "0.000"}}, {"currency_pair": "ftmusd", "market": "ftmusd", "fees": {"maker": "0.000", "taker": "0.000"}}, {"currency_pair": "galaeur", "market": "galaeur", "fees": {"maker": "0.000", "taker": "0.000"}}, {"currency_pair": "galausd", "market": "galausd", "fees": {"maker": "0.000", "taker": "0.000"}}, {"currency_pair": "gbpusd", "market": "gbpusd", "fees": {"maker": "0.000", "taker": "0.000"}}, {"currency_pair": "godseur", "market": "godseur", "fees": {"maker": "0.000", "taker": "0.000"}}, {"currency_pair": "godsusd", "market": "godsusd", "fees": {"maker": "0.000", "taker": "0.000"}}, {"currency_pair": "grteur", "market": "grteur", "fees": {"maker": "0.000", "taker": "0.000"}}, {"currency_pair": "grtusd", "market": "grtusd", "fees": {"maker": "0.000", "taker": "0.000"}}, {"currency_pair": "gusdusd", "market": "gusdusd", "fees": {"maker": "0.000", "taker": "0.000"}}, {"currency_pair": "hbareur", "market": "hbareur", "fees": {"maker": "0.000", "taker": "0.000"}}, {"currency_pair": "hbarusd", "market": "hbarusd", "fees": {"maker": "0.000", "taker": "0.000"}}, {"currency_pair": "imxeur", "market": "imxeur", "fees": {"maker": "0.000", "taker": "0.000"}}, {"currency_pair": "imxusd", "market": "imxusd", "fees": {"maker": "0.000", "taker": "0.000"}}, {"currency_pair": "injeur", "market": "injeur", "fees": {"maker": "0.000", "taker": "0.000"}}, {"currency_pair": "injusd", "market": "injusd", "fees": {"maker": "0.000", "taker": "0.000"}}, {"currency_pair": "knceur", "market": "knceur", "fees": {"maker": "0.000", "taker": "0.000"}}, {"currency_pair": "kncusd", "market": "kncusd", "fees": {"maker": "0.000", "taker": "0.000"}}, {"currency_pair": "ldoeur", "market": "ldoeur", "fees": {"maker": "0.000", "taker": "0.000"}}, {"currency_pair": "ldousd", "market": "ldousd", "fees": {"maker": "0.000", "taker": "0.000"}}, {"currency_pair": "linkbtc", "market": "linkbtc", "fees": {"maker": "0.000", "taker": "0.000"}}, {"currency_pair": "linkeur", "market": "linkeur", "fees": {"maker": "0.000", "taker": "0.000"}}, {"currency_pair": "linkgbp", "market": "linkgbp", "fees": {"maker": "0.000", "taker": "0.000"}}, {"currency_pair": "linkusd", "market": "linkusd", "fees": {"maker": "0.000", "taker": "0.000"}}, {"currency_pair": "lrceur", "market": "lrceur", "fees": {"maker": "0.000", "taker": "0.000"}}, {"currency_pair": "lrcusd", "market": "lrcusd", "fees": {"maker": "0.000", "taker": "0.000"}}, {"currency_pair": "ltcbtc", "market": "ltcbtc", "fees": {"maker": "0.000", "taker": "0.000"}}, {"currency_pair": "ltceur", "market": "ltceur", "fees": {"maker": "0.000", "taker": "0.000"}}, {"currency_pair": "ltcgbp", "market": "ltcgbp", "fees": {"maker": "0.000", "taker": "0.000"}}, {"currency_pair": "ltcusd", "market": "ltcusd", "fees": {"maker": "0.000", "taker": "0.000"}}, {"currency_pair": "manaeur", "market": "manaeur", "fees": {"maker": "0.000", "taker": "0.000"}}, {"currency_pair": "manausd", "market": "manausd", "fees": {"maker": "0.000", "taker": "0.000"}}, {"currency_pair": "maticeur", "market": "maticeur", "fees": {"maker": "0.000", "taker": "0.000"}}, {"currency_pair": "maticusd", "market": "maticusd", "fees": {"maker": "0.000", "taker": "0.000"}}, {"currency_pair": "mkreur", "market": "mkreur", "fees": {"maker": "0.000", "taker": "0.000"}}, {"currency_pair": "mkrusd", "market": "mkrusd", "fees": {"maker": "0.000", "taker": "0.000"}}, {"currency_pair": "mpleur", "market": "mpleur", "fees": {"maker": "0.000", "taker": "0.000"}}, {"currency_pair": "mplusd", "market": "mplusd", "fees": {"maker": "0.000", "taker": "0.000"}}, {"currency_pair": "neareur", "market": "neareur", "fees": {"maker": "0.000", "taker": "0.000"}}, {"currency_pair": "nearusd", "market": "nearusd", "fees": {"maker": "0.000", "taker": "0.000"}}, {"currency_pair": "nexoeur", "market": "nexoeur", "fees": {"maker": "0.000", "taker": "0.000"}}, {"currency_pair": "nexousd", "market": "nexousd", "fees": {"maker": "0.000", "taker": "0.000"}}, {"currency_pair": "paxusd", "market": "paxusd", "fees": {"maker": "0.000", "taker": "0.000"}}, {"currency_pair": "perpeur", "market": "perpeur", "fees": {"maker": "0.000", "taker": "0.000"}}, {"currency_pair": "perpusd", "market": "perpusd", "fees": {"maker": "0.000", "taker": "0.000"}}, {"currency_pair": "radeur", "market": "radeur", "fees": {"maker": "0.000", "taker": "0.000"}}, {"currency_pair": "radusd", "market": "radusd", "fees": {"maker": "0.000", "taker": "0.000"}}, {"currency_pair": "rlyeur", "market": "rlyeur", "fees": {"maker": "0.000", "taker": "0.000"}}, {"currency_pair": "rlyusd", "market": "rlyusd", "fees": {"maker": "0.000", "taker": "0.000"}}, {"currency_pair": "rndreur", "market": "rndreur", "fees": {"maker": "0.000", "taker": "0.000"}}, {"currency_pair": "rndrusd", "market": "rndrusd", "fees": {"maker": "0.000", "taker": "0.000"}}, {"currency_pair": "sandeur", "market": "sandeur", "fees": {"maker": "0.000", "taker": "0.000"}}, {"currency_pair": "sandusd", "market": "sandusd", "fees": {"maker": "0.000", "taker": "0.000"}}, {"currency_pair": "sgbeur", "market": "sgbeur", "fees": {"maker": "0.000", "taker": "0.000"}}, {"currency_pair": "sgbusd", "market": "sgbusd", "fees": {"maker": "0.000", "taker": "0.000"}}, {"currency_pair": "shibeur", "market": "shibeur", "fees": {"maker": "0.000", "taker": "0.000"}}, {"currency_pair": "shibusd", "market": "shibusd", "fees": {"maker": "0.000", "taker": "0.000"}}, {"currency_pair": "skleur", "market": "skleur", "fees": {"maker": "0.000", "taker": "0.000"}}, {"currency_pair": "sklusd", "market": "sklusd", "fees": {"maker": "0.000", "taker": "0.000"}}, {"currency_pair": "slpeur", "market": "slpeur", "fees": {"maker": "0.000", "taker": "0.000"}}, {"currency_pair": "slpusd", "market": "slpusd", "fees": {"maker": "0.000", "taker": "0.000"}}, {"currency_pair": "snxeur", "market": "snxeur", "fees": {"maker": "0.000", "taker": "0.000"}}, {"currency_pair": "snxusd", "market": "snxusd", "fees": {"maker": "0.000", "taker": "0.000"}}, {"currency_pair": "soleur", "market": "soleur", "fees": {"maker": "0.000", "taker": "0.000"}}, {"currency_pair": "solusd", "market": "solusd", "fees": {"maker": "0.000", "taker": "0.000"}}, {"currency_pair": "storjeur", "market": "storjeur", "fees": {"maker": "0.000", "taker": "0.000"}}, {"currency_pair": "storjusd", "market": "storjusd", "fees": {"maker": "0.000", "taker": "0.000"}}, {"currency_pair": "suieur", "market": "suieur", "fees": {"maker": "0.000", "taker": "0.000"}}, {"currency_pair": "suiusd", "market": "suiusd", "fees": {"maker": "0.000", "taker": "0.000"}}, {"currency_pair": "sushieur", "market": "sushieur", "fees": {"maker": "0.000", "taker": "0.000"}}, {"currency_pair": "sushiusd", "market": "sushiusd", "fees": {"maker": "0.000", "taker": "0.000"}}, {"currency_pair": "sxpeur", "market": "sxpeur", "fees": {"maker": "0.000", "taker": "0.000"}}, {"currency_pair": "sxpusd", "market": "sxpusd", "fees": {"maker": "0.000", "taker": "0.000"}}, {"currency_pair": "umaeur", "market": "umaeur", "fees": {"maker": "0.000", "taker": "0.000"}}, {"currency_pair": "umausd", "market": "umausd", "fees": {"maker": "0.000", "taker": "0.000"}}, {"currency_pair": "unibtc", "market": "unibtc", "fees": {"maker": "0.000", "taker": "0.000"}}, {"currency_pair": "unieur", "market": "unieur", "fees": {"maker": "0.000", "taker": "0.000"}}, {"currency_pair": "uniusd", "market": "uniusd", "fees": {"maker": "0.000", "taker": "0.000"}}, {"currency_pair": "usdceur", "market": "usdceur", "fees": {"maker": "0.000", "taker": "0.000"}}, {"currency_pair": "usdcusd", "market": "usdcusd", "fees": {"maker": "0.000", "taker": "0.000"}}, {"currency_pair": "usdcusdt", "market": "usdcusdt", "fees": {"maker": "0.000", "taker": "0.000"}}, {"currency_pair": "usdteur", "market": "usdteur", "fees": {"maker": "0.000", "taker": "0.000"}}, {"currency_pair": "usdtusd", "market": "usdtusd", "fees": {"maker": "0.000", "taker": "0.000"}}, {"currency_pair": "vegaeur", "market": "vegaeur", "fees": {"maker": "0.000", "taker": "0.000"}}, {"currency_pair": "vegausd", "market": "vegausd", "fees": {"maker": "0.000", "taker": "0.000"}}, {"currency_pair": "wbtcbtc", "market": "wbtcbtc", "fees": {"maker": "0.000", "taker": "0.000"}}, {"currency_pair": "xlmbtc", "market": "xlmbtc", "fees": {"maker": "0.000", "taker": "0.000"}}, {"currency_pair": "xlmeur", "market": "xlmeur", "fees": {"maker": "0.000", "taker": "0.000"}}, {"currency_pair": "xlmgbp", "market": "xlmgbp", "fees": {"maker": "0.000", "taker": "0.000"}}, {"currency_pair": "xlmusd", "market": "xlmusd", "fees": {"maker": "0.000", "taker": "0.000"}}, {"currency_pair": "xrpbtc", "market": "xrpbtc", "fees": {"maker": "0.000", "taker": "0.000"}}, {"currency_pair": "xrpeur", "market": "xrpeur", "fees": {"maker": "0.000", "taker": "0.000"}}, {"currency_pair": "xrpgbp", "market": "xrpgbp", "fees": {"maker": "0.000", "taker": "0.000"}}, {"currency_pair": "xrpusd", "market": "xrpusd", "fees": {"maker": "0.000", "taker": "0.000"}}, {"currency_pair": "xrpusdt", "market": "xrpusdt", "fees": {"maker": "0.000", "taker": "0.000"}}, {"currency_pair": "yfieur", "market": "yfieur", "fees": {"maker": "0.000", "taker": "0.000"}}, {"currency_pair": "yfiusd", "market": "yfiusd", "fees": {"maker": "0.000", "taker": "0.000"}}, {"currency_pair": "zrxeur", "market": "zrxeur", "fees": {"maker": "0.000", "taker": "0.000"}}, {"currency_pair": "zrxusd", "market": "zrxusd", "fees": {"maker": "0.000", "taker": "0.000"}}]
    bitstamp_test.go:886: json: cannot unmarshal object into Go struct field TradingFees.Fees of type []bitstamp.MakerTakerFees
```

Reply:
Noted, working on it